mobileApp/app/src/main/java/com/lebel/novelbinge/domain/NovelFileReader.kt

```diff
@@ -0,0 +1,38 @@
+package com.lebel.novelbinge.domain
+
+import android.content.Context
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import java.io.File
+
+class NovelFileReader {
+    companion object {
+        fun ReadAllTitles(context: Context): List<NovelData> {
+            val novelsFolder = File(context.getExternalFilesDir(null), "novels")
+
+            // Make sure the folder exists
+            novelsFolder.mkdirs()
```

Comment:
Create private function `ensureFolderCreated(novelsFolder)` instead of writing a comment.